source/server/connection_handler_impl.cc

```diff
@@ -224,58 +242,46 @@ ConnectionHandlerImpl::getBalancedHandlerByTag(uint64_t listener_tag) {
 
 Network::BalancedConnectionHandlerOptRef
 ConnectionHandlerImpl::getBalancedHandlerByAddress(const Network::Address::Instance& address) {
```

Comment:
also think about change this method name to getBalancedTCPHandlerByAddress, since this method is only used for TCP listener